subsys/bluetooth/shell/gatt.c

```diff
@@ -1294,7 +1328,7 @@ SHELL_STATIC_SUBCMD_SET_CREATE(gatt_cmds,
 	SHELL_CMD_ARG(unregister, NULL,
 		      "unregister pre-predefined test service",
 		      cmd_unregister_test_svc, 1, 0),
-	SHELL_CMD_ARG(notify, NULL, "[data]", cmd_notify, 1, 1),
+	SHELL_CMD_ARG(notify, NULL, "<handle> <str>", cmd_notify, 3, 0),
```

Comment:
could we have an arg for different data types:
e.g. `gatt notify [handle] bin deadbeef1337` will write those hex values and `gatt notify [handle] str "hello world"` will write the "hello world" string.

Reply:
or make two commands: notify-str and notify-bin. But I think it's gonna duplicate code a bit

Reply:
I would prefer if the shell handled this somehow. Maybe like \x"deadbeef1337"?

Reply:
Sending hex or string will work the same for us if using `hex2bin`. The receiving side can then choose whether or not to display the data as a string or hex.

Reply:
Updated PR: The command now takes a hex value, like other commands like `gatt write`.